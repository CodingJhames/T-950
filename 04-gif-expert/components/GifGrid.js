import React, { useEffect, useState } from 'react'
import { useFetchGifs } from '../hooks/useFectchGifs';
// import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';


export const GifGrid = ( {category}) => {


    const { data:images, loading } = useFetchGifs( category);

    // const [images, setImages] = useState([]);

    // useEffect( () => {
    //     getGifs( category)
    //         .then( imgs => setImages( imgs ) );
    // }, [ category ])


  return (
    
    <>
    <h3>{ category }</h3>

    { loading && <p>Loading</p>}

    {/* { loading ? 'Cargando...' : 'data cargada!'} */}

    {/* <div className='card-grid'>
    
    <ol>
        <li >title</li>    
        
        {  images.map( img => (
            
            <li key={image.id}>{image.title}</li>
            <GifGridItem 
            key={img.id}
            {...img} />
            
        ))}
    </ol>

    </div> */}

    <div className='card-grid'>
    
    <ol>
         
        
        {  images.map( img => (
            
            <GifGridItem 
            key={img.id}
            {...img} />
            
        ))}
    </ol>

    </div> 
    </>

  )
}
