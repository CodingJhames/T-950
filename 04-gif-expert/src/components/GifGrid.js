import React, { useEffect, useState } from 'react'


export const GifGrid = ( {category}) => {


    const [images, setImages] = useState([]);

    useEffect( () => {
        getGifs();
    }, [])


    const getGifs = async () => {

        const url ='https://api.giphy.com/v1/gifs/search?limit=10&q=slam+dunk&api_key=BsW6v8H74iBT4mTYmspQQnvmzK5DaDQ6'

        const resp = await fetch( url);

        const {data} = await resp.json();

        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images.downsized_medium.url
            }

        })

        console.log(gifs);
        setImages( gifs);
    }

  return (
    <>
    <h3>{ category }</h3>
    <ol>
        {/* <li >title</li>     */}
        
        {  images.map( image => (
            
            <li key={image.id}>{image.title}</li>
            
        ))}
    </ol>

    </>
  )
}
