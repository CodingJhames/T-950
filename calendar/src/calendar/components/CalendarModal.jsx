
import React, { useState } from 'react'
import { addHours } from 'date-fns/esm';
import Modal from 'react-modal';
import DatePicker, { registerLocale } from "react-datepicker";
import 'react-datepicker/dist/react-datepicker.css';
import { es } from 'date-fns/locale';
import { differenceInSeconds } from 'date-fns';

registerLocale('es', es );

const customStyles = {
    content: {
        top: '50%',
        left: '50%',
        right: 'auto',
        bottom: 'auto',
        marginRight: '-50%',
        transform: 'translate(-50%, -50%)',
    },
};

Modal.setAppElement("#root");



export const CalendarModal = () => {

    const [isOpen, setIsOpen] = useState( true );

    const [formValues, setFormValues] = useState({
        title:'James',
        notes:'Mejia',
        start: new Date(),
        end: addHours( new Date(), 2 ),
    })


    const onCloseModal = () => {
        console.log('modal cerrando');
        setIsOpen( false );
    }

    const onInputChange = ({target}) => {
        setFormValues({
            ...formValues,
            [target.name]: target.value
        })
    }

    const onDateChanged = ( event, changing ) => {
        setFormValues({
            ...formValues,
            [changing]: event
        })
    }

    const onSubmit = ( event ) => {
        event.preventDefault();

        const difference = differenceInSeconds(formValues.end, formValues.start );
        // console.log( difference );

        if( isNaN( difference ) || difference <= 0 ) {
            console.log('Error en las fechas');
            return;  
        }

        if( formValues.title.length <= 0 ) return;

        console.log( formValues );

        // cerrar modal
        // remover errores en pantalla

    }


  return (
    <Modal
    isOpen={ isOpen }
        onRequestClose={onCloseModal}
        style={customStyles}
        className="modal" 
        overlayClassName="modal-fondo"
        closeTimeoutMS={400}>
            <h1> Nuevo evento </h1>
<hr />
<form className="container" onSubmit={ onSubmit } >

    <div className="form-group mb-2">
        <label>Fecha y hora inicio</label>
        <DatePicker
            onChange={ ( event ) => onDateChanged( event, 'start') }
            selected={ formValues.start }
            className='form-control'
            dateFormat='Pp'
            showTimeSelect
            locale='es'
        />
    </div>

    <div className="form-group mb-2">
        <label>Fecha y hora fin</label>
        <DatePicker
            minDate={ formValues.start}
            onChange={ ( event ) => onDateChanged( event, 'end') }
            selected={ formValues.end }
            className='form-control'
            dateFormat='Pp'
            showTimeSelect
            locale='es'
        />
    </div>

    <hr />
    <div className="form-group mb-2">
        <label>Titulo y notas</label>
        <input 
            type="text" 
            className="form-control"
            placeholder="Título del evento"
            name="title"
            autoComplete="off"
            value={ formValues.title }
            onChange={onInputChange }
        />
        <small id="emailHelp" className="form-text text-muted">Una descripción corta</small>
    </div>

    <div className="form-group mb-2">
        <textarea 
            type="text" 
            className="form-control"
            placeholder="Notas"
            rows="5"
            name="notes"
            value={ formValues.notes }
            onChange={onInputChange }
        ></textarea>
        <small id="emailHelp" className="form-text text-muted">Información adicional</small>
    </div>

    <button
        type="submit"
        className="btn btn-outline-primary btn-block"
    >
        <i className="far fa-save"></i>
        <span> Guardar</span>
    </button>

</form>
    </Modal>
    )
}
