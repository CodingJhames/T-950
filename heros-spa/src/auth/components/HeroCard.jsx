

import React from 'react'
import { Link } from 'react-router-dom';

const HeroCard = ( {
        id,
        superhero,
        publisher,
        alter_ego,
        first_appearance,
        characters,
}) => {

    const heroImageUrl = `/assets/heroes/${id}.jpg`;

    const charactersByHero = ( <p>{characters}</p> );

    /* otra alternativa que sugiere fernando es crear un componente
    que sólo se use aquí, en razón que según él es más fácil de leer. 

    const charactersByHero = ( { alter_ego, characters  } ) => {

        if ( alter_ego === characters ) return (<></>);
            
        return <p>{characters}</p>; 
        
    } */


  return (
    <div className='col'>
        <div className='card'>

            <div className="row no-gutters">
                <div className="col-4">
                    <img src={heroImageUrl} className='card-img' alt={superhero} />
                </div>
            </div>

            <div className="card-8">
                
                <div className="card-body">

                    <h5 className='card-title'>{ superhero}</h5>
                    <p className='card-text'>{alter_ego}</p>
                    
                    {/* primera forma para que no se muestren 2 veces el nombre del personaje */}

                    {/* {
                        ( alter_ego !== characters ) && ( <p>{characters}</p> )
                    } */}

                    {/* segunda forma que sugiere fernando Herrera */}

                    {
                        ( alter_ego !== characters ) && charactersByHero 
                    }

                    {/* 
                    
                    { 
                        ( alter_ego !== characters ) && <p> { characters } </p>
                    }
                    
                    */}


                    {/* Forma alterna que sugiere fernando a través de un componente
                    
                    <charactersByHero  characters={ characters } alter_ego={ alter_ego } />

                    */}


                    <p className='card-text'>
                        <small className='text-muted'>{first_appearance}</small>
                    </p>

                    <Link to={`/hero/${id}`}>
                        Mas..
                    </Link>


                </div>
            </div>


        </div>
    </div>
  )
}

export default HeroCard