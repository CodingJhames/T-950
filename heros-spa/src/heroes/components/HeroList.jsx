
import React from 'react';
import { getHeroesByPublisher } from "../helpers/getHeroesByPublisher"


const HeroList = ( {publisher} ) => {
  
    const heroes = getHeroesByPublisher( publisher );
  
  
    return (
    
    <ul>
        {
            heroes.map( heroe =>(
                <li key={heroe.id}>{heroe.superhero}</li>
            ))
        }


    </ul>
  )
}

export default HeroList