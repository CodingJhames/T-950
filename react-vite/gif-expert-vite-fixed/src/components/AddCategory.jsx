import { useState } from "react"
import  PropTypes from "prop-types";


export const AddCategory = ( {onNewCategory } ) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ( {target} ) => {
        setInputValue(target.value);
        // console.log(target.value);
    }

    const onSubmit = (e) => {
        // console.log('Hola Parchita desde onSubmit');
        e.preventDefault();
        if( inputValue.trim().length <= 1 ) return;
        // setCategories( categories => [ inputValue,...categories ] );
        onNewCategory( inputValue.trim() );
        setInputValue('');

    }

  return (

    <form onSubmit={ e => onSubmit(e)} aria-label="form" >

        <input 
            type="text"
            placeholder="buscar Gifs" 
            value={inputValue}
            // onChange={ ( event ) => OnInputChange(event) }
            onChange={ onInputChange }
        />

    </form>

  )
}

AddCategory.propTypes = {

    onNewCategory: PropTypes.func.isRequired,
}

export default AddCategory