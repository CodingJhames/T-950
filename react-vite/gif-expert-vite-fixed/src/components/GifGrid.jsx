import { useEffect, useState } from "react";
// import { getGifs } from "../helpers/getGifs;"
import { useFecthGifs } from "../hooks/useFecthGifs";
import GifItem from "./GifItem";

    
  
    export const GifGrid = ( {category} ) => {
        

       const { images, isLoading } = useFecthGifs( category );

      //  console.log( images, isLoading );

      // const [images, setImages] = useState([]);

      // useEffect( () => {
      //   getGifs( category )
      //     .then( newImages => setImages( newImages) );
      // }, [] )

      
      return (
        <>

        <h3>{category}</h3>

        {
          isLoading ? 
           (<h2>Loading...</h2>)
          : null
        }

          <div className="card-grid">

            { images.map( (image) => (

              < GifItem 
              key={ image.id }
              { ...image } 
              />
              
            ) )
            }          
            
          </div>

          </>
      )
    }
    
    export default GifGrid