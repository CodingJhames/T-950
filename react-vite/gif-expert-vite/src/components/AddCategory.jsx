import { useState } from "react"


const AddCategory = () => {

    const [inputValue, setInputValue] = useState('Demon Slayer');

    const onInputChange = ( {target} ) => {
        setInputValue(target.value);
        // console.log(target.value);
    }

    const onSubmit = (e) => {
        e.preventDefault();
        console.log(inputValue);
    }

  return (

    <form onSubmit={ e => onSubmit(e)} >

        <input 
            type="text"
            placeholder="buscar Gifs" 
            value={inputValue}
            // onChange={ ( event ) => OnInputChange(event) }
            onChange={ onInputChange }
        />

    </form>

  )
}

export default AddCategory