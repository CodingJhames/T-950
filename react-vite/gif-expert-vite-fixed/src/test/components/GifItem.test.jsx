
import { render } from "@testing-library/react";
import { GifItem } from "../../components/GifItem";

describe('Pruebas en GifItem', () => { 

    const title = 'Demon Slayer';
    const url   = 'https://demon-slayer.com/demon-slayer.jpg';

    test('Debe de hacer match con el snapshot', () => { 

       const {container} = render( < GifItem title={title} url={url} />)
       expect( container ).toMatchSnapshot();

     }) 


 });