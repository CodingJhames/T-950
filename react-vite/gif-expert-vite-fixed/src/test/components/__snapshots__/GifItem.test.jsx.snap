// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Demon Slayer"
      src="https://demon-slayer.com/demon-slayer.jpg"
    />
    <p>
      Demon Slayer
    </p>
  </div>
</div>
`;
